common-util/Details/ServiceState/StepThreePayload/utils.jsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable no-underscore-dangle */
+import { ethers } from 'ethers';
+import {
+  GNOSIS_SAFE_CONTRACT,
+  MULTI_SEND_CONTRACT,
+} from 'common-util/AbiAndAddresses';
+import { safeMultiSend } from 'common-util/Contracts';
+
+const safeContracts = require('@gnosis.pm/safe-contracts');
+
+export const handleMultisigSubmit = async ({
+  multisig,
+  threshold,
+  agentInstances,
+  serviceOwner,
+  chainId,
+  handleStep3Deploy,
+  radioValue,
+}) => {
+  const multisigContract = new ethers.Contract(
+    multisig,
+    GNOSIS_SAFE_CONTRACT.abi,
+    ethers.getDefaultProvider('https://chain.staging.autonolas.tech/'),
+  );
+
+  const nonce = await multisigContract.nonce();
+
+  const callData = [];
+  const txs = [];
+
+  // Add the addresses, but keep the threshold the same
+  for (let i = 0; i < agentInstances.length; i += 1) {
+    callData[i] = multisigContract.interface.encodeFunctionData(
+      'addOwnerWithThreshold',
+      [agentInstances[i], 1],
+    );
+    txs[i] = safeContracts.buildSafeTransaction({
+      to: multisig,
+      data: callData[i],
+      nonce: 0,
+    });
+  }
+
+  callData.push(
+    multisigContract.interface.encodeFunctionData('removeOwner', [
+      agentInstances[0],
+      serviceOwner,
+      threshold,
+    ]),
+  );
+  txs.push(
+    safeContracts.buildSafeTransaction({
+      to: multisig,
+      data: callData[callData.length - 1],
+      nonce: 0,
+    }),
+  );
+
+  const multiSendContract = new ethers.Contract(
+    safeMultiSend[31337][0],
```

Comment:
Here it's hardcoded for the staging network, would be good to pass the current `chainId`